jsettlers.logic/src/jsettlers/logic/map/original/OriginalMapFileContentReader.java

```diff
@@ -0,0 +1,749 @@
+	}
+	
+	
+	//- Read Big-Ending INT from Buffer
+	public int readBEIntFrom(int offset) {
+		if (mapContent == null) return 0;
+		return  ((mapContent[offset  ] & 0xFF) << 0) |
+				((mapContent[offset+1] & 0xFF) << 8) |
+				((mapContent[offset+2] & 0xFF) << 16) |
+				((mapContent[offset+3] & 0xFF) << 24);
+	}
+	
+	//- Read Big-Ending 2 Byte Number from Buffer
+	public int readBEWordFrom(int offset) {
+		if (mapContent == null) return 0;
+		return  ((mapContent[offset  ] & 0xFF) << 0) |
+				((mapContent[offset+1] & 0xFF) << 8);
+	}
+	
+	//- read the Higher 4-Bit of the buffer
+	public int readHighNibbleFrom(int offset) {
+		if (mapContent == null) return 0;
+		return (mapContent[offset] >> 4) & 0x0F;
+	}
+	
+	//- read the Lower 4-Bit of the buffer
+	public int readLowNibbleFrom(int offset) {
+		if (mapContent == null) return 0;
+		return (mapContent[offset]) & 0x0F;
+		
+	}
+
+	//- read a C-Style String from Buffer (ends with the first \0)
+	public String readCStrFrom(int offset, int length) {
+		if (mapContent == null) return "";
+		if (mapContent.length <= offset + length) return "";
+		
+		String outStr = "";
+		int pos = offset;
+		
+		for (int i = length; i > 0; i--) {
+			byte b = mapContent[pos];
+			pos++;
+			
+			if (b == 0) break;
+			
+			outStr += new String(new byte[] {b}); 
+		}
+		return outStr;
+	}
+	
+	
+	//- returns a File Resources
+	private MapResourceInfo findResource(OriginalMapFileDataStructs.EMapFilePartType type) {
+		Iterator<MapResourceInfo> iterator = resources.iterator();
+		
+		while(iterator.hasNext()){
+			MapResourceInfo element = (MapResourceInfo) iterator.next();
+			
+			if (element.PartType == type) return element;
+		}
+		
+		return null;
+	}
+	
+	//- calculates the checksum of the file and compares it
+	boolean isChecksumValid() {
+		//- read Checksum from File
+		int fileChecksum = readBEIntFrom(0);
+		
+		mapData.fileChecksum = fileChecksum;
+		
+		//- make "count" a Multiple of four
+		int count = mapContent.length & 0xFFFFFFFC;
+		int currentChecksum = 0;
+		
+		//- Map Content start at Byte 8
+		for (int i = 8; i < count ; i+=4) {
+			//- read DWord
+			int currentInt = readBEIntFrom(i);
+			
+			//- using: Logic Right-Shift-Operator: >>>
+			currentChecksum = ((currentChecksum >>> 31) | ((currentChecksum << 1) ^ currentInt));
+		}
+
+		//- return TRUE if the checksum is OK!
+		return (currentChecksum == fileChecksum);
+	}
+	
+	
+	//- Reads in the Map-File-Structure
+	boolean loadMapResources() {
+		//- Version of File:  0x0A : Original Siedler Map ; 0x0B : Amazon Map
+		fileVersion = readBEIntFrom(4);
+		
+		//- check if the Version is compatible?
+		if ((fileVersion != EMapFileVersion.DEFAULT.value) && (fileVersion != EMapFileVersion.AMAZONS.value)) return false;
+		
+		//- Data lenght
+		int DataLenght = mapContent.length;
+
+		//- start of map-content
+		int FilePos = 8; 
+		int PartTypeTemp = 0;
+		
+		do {
+			PartTypeTemp = readBEIntFrom(FilePos);
+			int PartLen = readBEIntFrom(FilePos + 4);
```

Comment:
I fixed many camel case issues.

Unfortunately I can't promise that I found all issues because my IDE does not care about upper and lower case although my naming convention inspection is ```[a-z][A-Za-z\d]*```. When I search within one file I can check "Match Case" but not within the inspections :/